[inductor] Enable mypy checking for codegen/triton_foreach

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109643
* #109146

Summary: Add enough typehints to enable mypy checking for codegen/triton_foreach. Also fixed a bug in a dtype param.

Test Plan:
* `python test/inductor/test_foreach.py`
* `lintrunner`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

masnesral committed:
[inductor] Enable mypy checking for codegen/triton_foreach

Summary: Add enough typehints to enable mypy checking for codegen/triton_foreach. Also fixed a bug in a dtype param.

Test Plan:
* `python test/inductor/test_foreach.py`
* `lintrunner`

[ghstack-poisoned]

masnesral commented on torch/_inductor/codegen/triton_foreach.py:
```diff
@@ -144,10 +153,11 @@ def create_sub_kernel(self, *groups, index_dtype, mutations, reduction_hint):
         index_dtype = "tl.int32" if can_use_32bit else "tl.int64"
         _, _, signature = self.args.python_argdefs()
         triton_meta = {
-            "signature": signature_to_meta(signature, size_dtype=can_use_32bit),
```
mind vetting this part? `signature_to_meta` doesn't take a bool; it takes a type as a string.

mlazos replied:
I believe you can remove index_dtype and just use the size_dtype that you added, the values should be the same

masnesral commented:
@mlazos I started working on this while I was trying to fix `triton.py`; I didn't realize it was already claimed by you on https://github.com/pytorch/pytorch/issues/105230

mlazos reviewed (approved):
Everything looks good, just clean up the redundant vars

masnesral committed:
Update on "[inductor] Enable mypy checking for codegen/triton_foreach"


Summary: Add enough typehints to enable mypy checking for codegen/triton_foreach. Also fixed a bug in a dtype param.

Test Plan:
* `python test/inductor/test_foreach.py`
* `lintrunner`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
